functions/scripts/update-algolia-index.ts

```diff
@@ -0,0 +1,78 @@
+import fetch from 'node-fetch';
+import algoliasearch from 'algoliasearch';
+
+if (process.argv.length != 3) throw Error('Term argument not found.');
+
+const term: string = process.argv[2];
+
+if (!/20\d{2}0[1,5,9]/.test(term.trim()))
+  throw Error('Invalid term argument format');
+
+const BASE = 'https://courseup.vikelabs.ca/api';
+
+const client = algoliasearch('ALGOLIA_APP_ID', 'ALGOLIA_ADMIN_KEY');
+const index = client.initIndex('ALGOLIA_INDEX_NAME');
+
+interface Course {
+  subject: string;
+  code: string;
+  pid: string;
+  description: string;
+  title: string;
+}
+
+const getCourses = async (term: string): Promise<Course[]> => {
+  const res = await fetch(`${BASE}/courses/${term}`);
+  const data: Course[] = await res.json();
+  return data;
+};
+
+const getCourse = async (term: string, course: Course): Promise<Course> => {
+  const res = await fetch(
+    `${BASE}/courses/${term}/${course.subject}/${course.code}`
+  );
+  const data: Course = await res.json();
+  return data;
+};
+
+const main = async () => {
+  const courses: Course[] = await getCourses(term);
+
+  console.log(`Found ${courses.length} courses for ${term}`);
+
+  const values = courses.values();
+
+  const vals: Course[] = [];
+
+  const workers = Array(50)
+    .fill(values)
+    .map(async (iterator) => {
+      for (const course of iterator) {
+        const courseDetails = await getCourse(term, course);
+        console.log(course.subject, course.code);
+        vals.push({ ...courseDetails });
+      }
+    });
+
+  await Promise.allSettled(workers);
+
+  const coursesForIndex = vals.map((course) => ({
+    objectID: `${course.subject}${course.code}`,
+    subject: course.subject,
+    code: course.code,
```

Comment:
Having a field that's just the concatenation like `objectID` helps with cases where people enter `CSC100` without a space.

Reply:
is `objectID` not taken into consideration for the search?

Reply:
and re: synonyms for courses, are you thinking a sort of look up table with pre defined synonyms for the most common stuff?